pyblish/logic.py

```diff
@@ -153,10 +153,26 @@ def plugins_by_families(plugins, families):
 
     """
 
+    algorithms = {
```

Comment:
Since the same algorithms are being used in both this function, and `instances_by_plugin`, what do you think about moving this dictionary to the module-level, and prefix it with an underscore to highlight the fact that it is internal. That way, we could use it in both functions, and changing it would be simpler and more predictable.